process/transaction/transactionCostEstimator.go

```diff
@@ -74,10 +74,24 @@ func NewTransactionCostEstimator(
 	return tce, nil
 }
 
+// SimulateTransactionExecution will simulate a transaction's execution and will return the results
+func (tce *transactionCostEstimator) SimulateTransactionExecution(tx *transaction.Transaction) (*txSimData.SimulationResults, error) {
+	tce.mutExecution.Lock()
```

Comment:
:-1: this ensures all API calls will execute sequentially. The results should be correct in this way.